src/sidebar/util/test/group-organizations-test.js

```diff
@@ -0,0 +1,158 @@
+      const org2 = orgFixtures.organization({ name: 'Zirconium' });
+      const groups = [
+        { name: 'Aluminum', organization: org },
+        { name: 'Beryllium', organization: org2},
+        { name: 'Butane', organization: org2 },
+        { name: 'Cadmium', organization: org},
+      ];
+
+      const sortedGroups = groupsByOrganization(groups);
+
+      assert(sortedGroups[0].logo === org.logo);
+      assert(typeof sortedGroups[1].logo === 'undefined');
+      assert(sortedGroups[2].logo === org2.logo);
+      assert(typeof sortedGroups[3].logo === 'undefined');
+    });
+  });
+
+  context('when encountering missing data', function () {
+
+    it('should be able to sort without any groups in the default org', function () {
+      const org = orgFixtures.organization({ name:'Aluminum' });
+      const groups = [
+        { name: 'Aluminum', organization: org },
+        { name: 'Beryllium', organization: org},
+        { name: 'Cadmium', organization: org},
+      ];
+
+      const sortedGroups = groupsByOrganization(groups);
+
+      sortedGroups.forEach((group, idx) => {
+        assert.equal(group.name, groups[idx].name);
+      });
+
+    });
+
+    it('should omit any groups without an organization', function () {
+      const org = orgFixtures.organization({ name:'Europium' });
+      const groups = [
+        { name: 'Aluminum', organization: org },
+        { name: 'Beryllium', organization: org},
+        { name: 'Butane' },
+        { name: 'Cadmium', organization: org},
+      ];
+
+      const sortedGroups = groupsByOrganization(groups);
+
+      assert.equal(sortedGroups.length, groups.length - 1);
+      sortedGroups.forEach(group => {
+        assert.notEqual(group.name, 'Butane');
+      });
+    });
+
+    it('should omit any groups with unexpanded organizations', function () {
+      const org = orgFixtures.organization({ name:'Europium' });
+      const groups = [
+        { name: 'Aluminum', organization: org },
+        { name: 'Beryllium', organization: org},
+        { name: 'Butane', organization: 'foobar' },
+        { name: 'Cadmium', organization: org},
+      ];
+
+      const sortedGroups = groupsByOrganization(groups);
+
+      assert.equal(sortedGroups.length, groups.length - 1);
+      sortedGroups.forEach(group => {
+        assert.notEqual(group.name, 'Butane');
+      });
+
+    });
+
+    it ('should omit logo property if not present on organization', function () {
+      const org = orgFixtures.organization({ logo: undefined });
+      const org2 = orgFixtures.organization({ logo: null });
+      const groups = [
+        { name: 'Aluminum', organization: org },
+        { name: 'Beryllium', organization: org2},
+        { name: 'Butane', organization: org2 },
+        { name: 'Cadmium', organization: org},
+      ];
+
+      const sortedGroups = groupsByOrganization(groups);
+
+      sortedGroups.forEach(group => {
+        assert(typeof group.logo === 'undefined');
```

Comment:
I would recommend using `assert.equal` over `assert(bool expr)` for better error messages.